File: java/fundamentals/day05/MiniChallenge.java
package fundamentals.day05;
import java.util.Scanner;
public class MiniChallenge {
	

	    public static double[] getMarks(Scanner scanner, int count) {
	        double[] marks = new double[count];
	        System.out.println("Enter " + count + " subject marks:");
	        for (int i = 0; i < count; i++) {
	            System.out.print("Subject " + (i + 1) + ": ");
	            marks[i] = scanner.nextDouble();
	        }
	        return marks;
	    }

	    public static double calculateTotal(double[] marks) {
	        double total = 0;
	        for (double mark : marks) {
	            total += mark;
	        }
	        return total;
	    }

	    public static double calculateAverage(double total, int count) {
	        return total / count;
	    }

	    public static double findHighest(double[] marks) {
	        double highest = marks[0];
	        for (double mark : marks) {
	            if (mark > highest) highest = mark;
	        }
	        return highest;
	    }

	    public static double findLowest(double[] marks) {
	        double lowest = marks[0];
	        for (double mark : marks) {
	            if (mark < lowest) lowest = mark;
	        }
	        return lowest;
	    }

	    public static String calculateGrade(double average, double[] marks) {
	        for (double mark : marks) {
	            if (mark < 35) return "F (Failed standard)";
	        }
	        if (average >= 90) return "A+";
	        if (average >= 80) return "A";
	        if (average >= 70) return "B";
	        if (average >= 60) return "C";
	        if (average >= 50) return "D";
	        return "F";
	    }

	    public static void printReport(String name, double total, double average, double highest, double lowest, String grade) {
	        System.out.println("\n==============================");
	        System.out.println("      STUDENT REPORT CARD      ");
	        System.out.println("==============================");
	        System.out.println("Name    : " + name);
	        System.out.printf("Total   : %.2f%n", total);
	        System.out.printf("Average : %.2f%n", average);
	        System.out.printf("Highest : %.2f%n", highest);
	        System.out.printf("Lowest  : %.2f%n", lowest);
	        System.out.println("Grade   : " + grade);
	        System.out.println("==============================");
	    }

	    public static void main(String[] args) {
	        Scanner scanner = new Scanner(System.in);

	        System.out.print("Enter Student Name: ");
	        String name = scanner.nextLine();

	        double[] marks = getMarks(scanner, 5);
	        double total = calculateTotal(marks);
	        double average = calculateAverage(total, marks.length);
	        double highest = findHighest(marks);
	        double lowest = findLowest(marks);
	        String grade = calculateGrade(average, marks);

	        printReport(name, total, average, highest, lowest, grade);

	        scanner.close();
	    }
	}

File: python/fundamentals/day-05/MiniChallenge.py
def get_marks(count=5):
    marks = []
    print(f"Enter {count} subject marks:")
    for i in range(1, count + 1):
        marks.append(float(input(f"Subject {i}: ")))
    return marks

def calculate_total(marks):
    return sum(marks)

def calculate_average(total, count):
    return total / count

def find_highest(marks):
    return max(marks)

def find_lowest(marks):
    return min(marks)

def calculate_grade(average, marks):
    # If any mark is below passing (35), student fails
    if any(m < 35 for m in marks):
        return "F (Failed standard)"
    if average >= 90: return "A+"
    elif average >= 80: return "A"
    elif average >= 70: return "B"
    elif average >= 60: return "C"
    elif average >= 50: return "D"
    else: return "F"

def print_report(name, total, average, highest, lowest, grade):
    print("\n" + "=" * 30)
    print("      STUDENT REPORT CARD      ")
    print("=" * 30)
    print(f"Name    : {name}")
    print(f"Total   : {total:.2f}")
    print(f"Average : {average:.2f}")
    print(f"Highest : {highest:.2f}")
    print(f"Lowest  : {lowest:.2f}")
    print(f"Grade   : {grade}")
    print("=" * 30)

def main():
    name = input("Enter Student Name: ")
    marks = get_marks(5)
    
    total = calculate_total(marks)
    average = calculate_average(total, len(marks))
    highest = find_highest(marks)
    lowest = find_lowest(marks)
    grade = calculate_grade(average, marks)
    
    print_report(name, total, average, highest, lowest, grade)

if __name__ == "__main__":
    main()